src/main/java/ir/sahab/zookeeperrule/ZooKeeperRule.java

```diff
@@ -29,28 +30,41 @@
     private String localIp;
 
     public ZooKeeperRule() {
-        this(anOpenPort());
+        this(constructZkAddress());
     }
 
-    public ZooKeeperRule(int port) {
-        if (port < 0) {
-            throw new IllegalArgumentException("Port number must be positive");
-        }
-        this.port = port;
-    }
-
-    @Override
-    protected void before() throws IOException, InterruptedException {
-        snapshotDir = Files.createTempDirectory("zk-snapshot").toFile();
-        logDir = Files.createTempDirectory("zk-logs").toFile();
+    private static String constructZkAddress() {
 
         // Why we are going to use local IP and not just localhost or 127.0.0.1 constants?
         // Because we have encountered a problem when configured an KafkaServerStartable
         // to use this embedded ZooKeeper on 'localhost'.
         // But using local IP, solved the problem. See this:
         // https://www.ibm.com/support/knowledgecenter/SSPT3X_4.1.0/
         // com.ibm.swg.im.infosphere.biginsights.trb.doc/doc/trb_kafka_producer_localhost.html
-        localIp = InetAddress.getLocalHost().getHostAddress();
+        String localIp = null;
+        try {
+            localIp = InetAddress.getLocalHost().getHostAddress();
+        } catch (UnknownHostException e) {
+            e.printStackTrace();
+        }
+
+        return localIp + ":" + anOpenPort();
+    }
+
+    public ZooKeeperRule(String address) {
```

Comment:
Add javadoc:
Creates a rule which will setup an embedded ZooKeeper on the given address. 
@param address the local address on which the embedded ZooKeeper should be setup. It should be in format of "IP:PORT" and the IP should be one of the local IPs of the local system.